homeassistant/components/matter/lock.py

```diff
@@ -79,11 +79,27 @@ def _update_from_device(self) -> None:
 
         if lock_state is clusters.DoorLock.Enums.DlLockState.kLocked:
             self._attr_is_locked = True
+            self._attr_is_locking = False
+            self._attr_is_unlocking = False
         elif lock_state is clusters.DoorLock.Enums.DlLockState.kUnlocked:
             self._attr_is_locked = False
+            self._attr_is_locking = False
+            self._attr_is_unlocking = False
+        elif lock_state is clusters.DoorLock.Enums.DlLockState.kNotFullyLocked:
+            if self._attr_is_locked is True:
+                self._attr_is_unlocking = True
+            elif self._attr_is_locked is False:
+                self._attr_is_locking = True
         else:
             self._attr_is_locked = None
-            LOGGER.warning("Unknown lock state: %s for %s", lock_state, self.entity_id)
+            self._attr_is_locking = None
+            self._attr_is_unlocking = None
+            # According to the matter docs a null state can happen during device startup.
+            LOGGER.warning(
```

Comment:
I don't think we need a warning. Unknown state is a known concept in Home Assistant. Keep the comment but move it above the state attribute modification in this block.